packages/datagateway-search/src/table/datasetSearchTable.component.test.tsx

```diff
@@ -1,662 +1,577 @@
-    expect(history.length).toBe(2);
-    expect(history.location.search).toBe(
-      `?filters=${encodeURIComponent(
-        '{"name":{"value":"test","type":"include"}}'
-      )}`
-    );
+    renderComponent();
 
-    filterInput.instance().value = '';
-    filterInput.simulate('change');
+    // wait for data to finish loading
+    expect(await screen.findByText('Dataset test name')).toBeInTheDocument();
 
-    expect(history.length).toBe(3);
-    expect(history.location.search).toBe('?');
-  });
+    // pretend the server has removed the item from the cart
+    // create a new array to trigger useMemo update
+    cartItems = [];
 
-  it('updates filter query params on date filter', () => {
-    applyDatePickerWorkaround();
+    // clicks on the row checkbox
+    await user.click(screen.getByRole('checkbox', { name: 'select row 0' }));
 
-    const wrapper = createWrapper();
-
-    const filterInput = wrapper.find(
-      'input[id="datasets.modified_time filter to"]'
-    );
-    filterInput.instance().value = '2019-08-06';
-    filterInput.simulate('change');
-
-    expect(history.length).toBe(2);
-    expect(history.location.search).toBe(
-      `?filters=${encodeURIComponent('{"modTime":{"endDate":"2019-08-06"}}')}`
-    );
-
-    filterInput.instance().value = '';
-    filterInput.simulate('change');
-
-    expect(history.length).toBe(3);
-    expect(history.location.search).toBe('?');
-
-    cleanupDatePickerWorkaround();
+    // the checkbox should be checked
+    expect(
+      await screen.findByRole('checkbox', { name: 'select row 0' })
+    ).not.toBeChecked();
   });
 
-  it('updates sort query params on sort', () => {
-    const wrapper = createWrapper();
-
-    wrapper
-      .find('[role="columnheader"] span[role="button"]')
-      .first()
-      .simulate('click');
+  it('selected rows only considers relevant cart items', async () => {
+    cartItems = [
+      {
+        entityId: 1,
+        entityType: 'investigation',
+        id: 1,
+        name: 'test',
+        parentEntities: [],
+      },
+      {
+        entityId: 2,
+        entityType: 'dataset',
+        id: 2,
+        name: 'test',
+        parentEntities: [],
+      },
+    ];
 
-    expect(history.length).toBe(2);
-    expect(history.location.search).toBe(
-      `?sort=${encodeURIComponent('{"name":"asc"}')}`
-    );
-  });
+    renderComponent();
 
-  it('calls addToCart mutate function on unchecked checkbox click', () => {
-    const addToCart = jest.fn();
-    (useAddToCart as jest.Mock).mockReturnValue({
-      mutate: addToCart,
-      loading: false,
+    const selectAllCheckbox = await screen.findByRole('checkbox', {
+      name: 'select all rows',
     });
-    const wrapper = createWrapper();
 
-    wrapper.find('[aria-label="select row 0"]').last().simulate('click');
-
-    expect(addToCart).toHaveBeenCalledWith([1]);
+    expect(selectAllCheckbox).not.toBeChecked();
+    expect(selectAllCheckbox).toHaveAttribute('data-indeterminate', 'false');
```

Comment:
Is this right? If only some checkboxes are checked, surely the selectAllCheckbox is in an indeterministic state, right?

Reply:
Ahh okay, thanks for the clarification